### PR TITLE
Use fbjni from maven

### PR DESCRIPTION
**Pull request by kirklandsign**
TL;DR: We need to make sure the runtime so deps is the same as compile time so library. We can't build the so from source, because runtime uses another one.

Instead of importing fbjni repo into executorch, and building from executorch as a subdir, we need to use a pre-built (on maven repo) fbjni.so together with its headers. We need to link our executorch JNI library against that fbjni.so. We can't link against fbjni as a static library, because the Java layer needs to load it, and we can't have duplicated symbol from static and shared lib.

For Android apk developer, they need to import executorch as AAR, as well as fbjni AAR from maven. Inside fbjni AAR, we don't know which NDK is the fbjni.so built. If the NDK version between executorch and fbjni is different, we may have incompatible c++ library. This caused issues like 
```
java.lang.UnsatisfiedLinkError: dlopen failed: cannot locate symbol "_ZTVNSt6__ndk114basic_ifstreamIcNS_11char_traitsIcEEEE" referenced by "/data/app/~~enBXzd5mY8qQNSRdnGW4vg==/com.example.executorchllamademo-u8anctHQVTL1jVJsZsEqRQ==/lib/arm64/libexecutorch.so"...
``` 
in the past.

Therefore, we need to always make sure we have a specific fbjni version for a given executorch release (say 0.5.1 for 20241010). We download the AAR from maven, which provides "prefab" artifacts, including the so library and header for us to compile ET jni code.

With this change, we can build ET JNI with any NDK version, regardless of how fbjni built their library.

### CODE REVIEW

File: extension/android/CMakeLists.txt [hunk 1]
```diff
@@ -21,9 +21,40 @@ include(${EXECUTORCH_ROOT}/build/Utils.cmake)
...
+# to ensure the libc++ dependencies are consistent.
+# WARNING #
+# Users need to use the SAME fbjni version here and in app gradle dependency
+# for runtime compatibility!
+if(NOT FBJNI_VERSION)
+  set(FBJNI_VERSION 0.5.1)
+endif()
+
+set(FBJNI_AAR_URL https://repo1.maven.org/maven2/com/facebook/fbjni/fbjni/${FBJNI_VERSION}/fbjni-${FBJNI_VERSION}.aar)
+set(FBJNI_DOWNLOAD_PATH ${CMAKE_CURRENT_BINARY_DIR}/third-party/fbjni/fbjni.aar)
```
Comment (larryliu0820) [inline extension/android/CMakeLists.txt:37]: This seems risky? How do you make sure things under `third-party/fbjni` is compatible with `fbjni.aar` downloaded from Maven?

Comment (kirklandsign) [reply extension/android/CMakeLists.txt:37]: Sorry I need to remove examples/third-party/fbjni in a follow up PR. We can't use it. We need to make sure the runtime so deps is the same as compile time so library. We can't build the so from source, because runtime uses another one. The `fbjni.aar` downloaded from Maven basically contains a "SDK" of headers and so library for us to link our ET library against.

**Review by larryliu0820 — APPROVED** [review]
